src/context/internal.rs

```diff
@@ -116,6 +129,96 @@ pub trait ContextInternal<'a>: Sized {
 
     fn activate(&self) { self.scope_metadata().active.set(true); }
     fn deactivate(&self) { self.scope_metadata().active.set(false); }
+
+    #[cfg(feature = "legacy-runtime")]
+    fn try_catch_internal<'b: 'a, F>(&mut self, f: F) -> Result<Handle<'a, JsValue>, Handle<'a, JsValue>>
+        where F: UnwindSafe + FnOnce(&mut Self) -> JsResult<'b, JsValue>
+    {
+        // A closure does not have a guaranteed layout, so we need to box it in order to pass
+        // a pointer to it across the boundary into C++.
+        let rust_thunk = Box::into_raw(Box::new(f));
+
+        let mut local: MaybeUninit<raw::Local> = MaybeUninit::zeroed();
+        let mut unwind_value: MaybeUninit<*mut c_void> = MaybeUninit::zeroed();
```

Comment:
I think `MaybeUninit::uninit()` may be preferred.

> Incorrect usage of this function: initializing a struct with zero, where some fields cannot hold 0 as a valid value.

https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#zeroed.v

Reply:
The docs are a little misleading here: it's not _incorrect_ usage, it's just not a valid initialization to use zero bits if that's not a valid representation of `T`. The undefined behavior only kicks in if you screw up the `.assume_init()` assertion. But it's still strictly safer in practice to always use `::zeroed()`, because at least your undefined behavior was based on a zeroed-out memory slot rather than whatever bits happened to be in that slot from before.

TL;DR A good defense-in-depth policy is always to use `MaybeUninit::zeroed()` when using the FFI to initialize a slot, even though it's still your responsibility to ensure `.assume_init()` is being used correctly.